src/NuGet.Core/NuGet.Shared/TypeExtensions.cs

```diff
@@ -0,0 +1,23 @@
+#if IS_NET40_CLIENT
+using System;
+
+namespace System.Reflection
+{
+    /// <summary>
+    /// Extension methods for the <see cref="Type"/> type.
+    /// </summary>
+    internal static class TypeExtensions
+    {
+        /// <summary>
+        /// Expose the type itself as the type info. This is to make TypeInfo available on
+        /// .NET Framework 4 Client Profile. This method is used in the automatically generated
+        /// resource C# files. To avoid manual changes to an automatically generated file, expose
+        /// this method as an extension method in an used namespace.
+        /// </summary>
+        public static Type GetTypeInfo(this Type type)
```

Comment:
I would add a TypeInfo type instead of returning Type directly. And add whatever wrapping methods you need